src/server.py

```diff
@@ -1,26 +1,57 @@
 """Module to set up the server."""
+# encoding: utf-8
 import socket
+import email.utils
 
 
 def server():
     """Start server to receive message and echo back."""
     server = socket.socket(socket.AF_INET,
                            socket.SOCK_STREAM,
                            socket.IPPROTO_TCP)
-    address = ("127.0.0.1", 5011)
+    address = ("127.0.0.1", 5000)
     server.bind(address)
     server.listen(1)
     buffer_length = 8
     while True:
         try:
             conn, addr = server.accept()
-            message = ""
-            while message[-2:] != "\r\n":
-                message += conn.recv(buffer_length).decode("utf8")
-            conn.sendall(message.encode("utf8"))
+            request = ""
+            while request[-4:] != "\r\n\r\n":
+                request += conn.recv(buffer_length).decode("utf8")
```

Comment:
Couple of improvements to be made here.

First, in terms of efficiency, it is much better in Python to append chunks of strings to a list, then use the str.join method to join them all at the end, rather than concatenate one at a time like this.

Secondly, this will break sometimes on non-unicode characters. So you're decoding each chunk as you get it. Since unicode characters can take up multiple bytes, what happens when a multiple-byte unicode character isn't received all the way in one chunk? i.e. the buffer length cuts it in half?

Rather than decoding one chunk at a time, what would be a better approach?